src/ansi-c/library/stdlib.c

```diff
@@ -115,23 +115,31 @@ inline void *malloc(__CPROVER_size_t malloc_size)
   // realistically, malloc may return NULL,
   // and __CPROVER_allocate doesn't, but no one cares
   __CPROVER_HIDE:;
-  void *malloc_res;
-  malloc_res = __CPROVER_allocate(malloc_size, 0);
 
-  // make sure it's not recorded as deallocated
-  __CPROVER_deallocated=(malloc_res==__CPROVER_deallocated)?0:__CPROVER_deallocated;
+    __CPROVER_assert(
+      __CPROVER_malloc_may_fail == (__CPROVER_bool)0, "malloc may fail");
```

Comment:
? Why is there an assert that this is false? I thought this would be more like `if (__CPROVER_malloc_may_fail && __VERIFIER_nondet___CPROVER_bool()) return NULL;`

Reply:
Because conditionally returning `NULL` breaks too many things down the line in CBMC.

Reply:
I take it back. Returning `NULL` seems to work here just fine.